### ISSUE TITLE
POP3 connection attempts loop out of a sudden

### ISSUE THREAD

**Issue by d-albrecht** [thunderbird/thunderbird-android#5884]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

5.600

### Where did you get the app from?

F-Droid

### Android version

7.1.2 (and I guess that would count as custom ROM)

### Device model

Fairphone model 2

### Steps to reproduce

(I don't really have steps to reproduce (other than "try to update the mail list via POP3") as this isn't a problem that occurred always when doing X but an issue that occurred over time)

Yesterday about that time my phone app was still working just fine. I have it set to fetch mails on a 3-hour schedule and besides some minor occurrences where the app needed about 15 seconds to finish the update of two accounts (which already is long from my point of view) I never had any problems with k9. But yesterday at about 6 pm I unlocked my phone to see the usual notification about mail updates being performed but after what I would estimate as two or three minutes the notification still didn't go away and was still on the first account (with no new mails to fetch as I checked aside from k9). So, not having any other method to stop this action, I force stopped k9 (several times in fact in my debugging process), at some point disabled the scheduled mail update (as k9 would only get stuck in this process again as soon as I opened the app again), cleared the cache several times, restarted the phone (one time even in safe mode where the app obviously can't work).

At some point I considered that this might be a server connection problem, so I opened up the account settings to (re-)configure the incoming and outgoing servers. **Is it correct that if I enter these sub-settings and click "Next", the server connection is (re-)validated and depending on the outcome of this action I can verify the state of the connection?** In the following I assume that's how that works, as (re-)configuring the incoming server had the same "problem" than updating the mail list with the benefit that this process is user-cancel-able whereas mail polling isn't.

What I know so far (please ask if you have any other ideas of how to pinpoint the exact issue better):
- I always use POP3 on all devices. That means that I can state for a fact that the POP3 connection in theory can work as Thunderbird can still retrieve my mails from the same accounts.
- I haven't checked in detail but I'm also almost certain that all my POP3 clients use the same setup when it comes to the server-connection. If that is in doubt and relevant, I can for sure check but I don't want to go down this rabbit hole right now.
- I haven't changed any settings in my mail account or k9 around the same time. In fact I think the last time I changed any settings was around the time I configured k9 on my phone which is about a year ago.
- The issue occurred suddenly. One mail update worked, the next one looped indefinitely without any error.
- I haven't yet tried to reinstall the app or clear the app data. I still hope that there will be some way to fix this without having to setup my accounts from scratch (which isn't hard but tedious).
- In the beginning of this issue sending test-mails still worked. As I describe in the "Actual behavior"-segment, even the outgoing connection isn't 100% reliable anymore.

### Expected behavior

Any attempts to contact a POP3-server should either be successful within seconds or provide some form of error message. Even a time out with some kind of feedback would be more helpful than the app retrying for literally minutes.

### Actual behavior

- Re-configuring the Outgoing server loops sometimes.
Either it finishes right away and sends me back to the "Sending Mails"-section of the account settings or it takes forever. Either way I can't identify a single message in the log to indicate that I even attempted to reconfigure the Outgoing server or the status of this action.

- Re-configuring the Incoming server loops all the time.
Depending on the time I give it to (re-)try and observing the logs I can usually see one (or up to three) Debug message(s) "OK POP server ready" but other than that I don't get any feedback on the status of the re-configuring besides that it apparently never finishes. Doing this on mobile data showed that k9 is actually sending data (several MB if you keep retrying for long enough). The message that is shown while k9 evaluates the connection isn't the same all the time. Currently I have "Checking incoming server settings" or "Retrieving account information" at a somewhat random basis, I don't see a pattern yet.

- Attempting to update the mail list loops forever without any feedback. Letting the app update the mail list on a schedule results in the situation that a notification appears that never resolves itself (keeps on turning).

### Logs

(I can provide some later, but at this time I just wanted this bug report to be out in public to (a) get some feedback on how to pinpoint the issue and (b) let the developers know that there is a problem in the first place)

**Comment by cketti** [comment]
K-9 Mail 5.600 is no longer supported.

If your provider is IONOS or GMX see #5879.

**Comment by d-albrecht** [comment]
Then I apparently have to leave k9 behind and find a fork that still uses the superior old UI with the newest features as well as security and bug fixes. Thanks for the quick reply, though.

PS: not supporting the second to last stable version anymore is a dubious decision especially now that you have done such a huge switch like replacing the whole UI. Open source software is about choice and what you're effectively doing is **not** offering choice but forcing users to accept certain things as given. If I wanted that I would (still) use Windows or Mac. That's what causes segregation in the open source world. But thankfully, forks exists for these exact situations.

PPS: And yes, I very much acknowledge that these mail providers are also to blame here and it's not purely k9's fault. But switching my mail client is something I can do without anyone around me noticing, switching my mail address and provider only to have an address that is still compatible with k9 would cause additional external changes I'm not willing to set off. In this regard you are the weak link in this chain.

**Comment by cketti** [comment]
> Open source software is about choice

You have the freedom to use the source code and make whatever changes you like. Open source doesn't mean someone else will do whatever you want them to.

**Comment by d-albrecht** [comment]
TL;DR: Yes, in an ideal world it would be that easy, but now that you got me started...

1.) Not everyone can code and even those who can might not be familiar with the Android ecosystem. So, true, forks can exist, but having/finding someone **able** to keep a fork alive isn't a trivial task.

2.) Take Firefox and Android/AOSP as examples. They are open source, but that doesn't mean that they are easily forkable. On the one hand there are several AOSP-forks that (all individually) spend time on trying to de-Google the upstream repo. Because open source doesn't mean that there is no tracking and such. That shows that having a clean basis would be up to the original developers, but no such clean basis projects exist as far as I know for software of this scale. On the other hand, I saw with Firefox time and time again that their commits aren't adaptable to different UIs (just ask the Waterfox devs how many commits they are behind for their Classic version). That's just another example for why open source isn't the same as having a clean basis to work with.

3.) Who wanted the new UI? Have you any numbers on this? Again referring to Firefox, back then I had to accept several UI re-designs that all made the product worse (before I finally switched browsers). And by the backlash you got for the 5.700 release it's apparent that some significant portion of your user base agrees with me. Have you seen rising Play Store download numbers since this release? Have you asked your users if they wanted a redesign? And don't get me started with your argument that the beta version with this new UI was available for a long time and no-one disliked it. Honestly, who uses beta versions for software that **has to work** all the time. And who would without any external nudges just out of curiosity install the beta version if they didn't know that there is a new design concept to review.

So, unless you prove that your base code is still compatible with the old UI (by maintaining two versions with both UIs yourself), you can't just point to your license and say "just do it if you want". Your open-source-ness is just a license-argument if the accessibility is as good/bad as it is with the other two mentioned open source projects. Open source also says something about your community. Or at least it should. Again, if you can prove to me that your users actually wanted this redesign in the majority, then I'm the odd one out.

And as a side note: You haven't done so here, but in the linked issue/forum thread you mentioned that IMAP is quote "a far better choice". Can you elaborate, why that is from your point of view? Because to me these two protocols are way too different to do a direct comparison. Sure, one might be older than the other and have other (fewer) benefits, but still POP3 is just what I want/need. No constant server connection (no constant battery drain), no immediate notification if some (spam-)mail arrives, no synchronization of my whole account, just the inbox (if you have more than one mail client you have the choice of managing all your mails on the server but then these are mostly legacy mails that you don't need synced or you have one client that pulls and deletes all mails, but then you can't really "work" in both clients, having several clients that all just sync the inbox is the best approach for me). And yes, it might be possible to configure IMAP to only do what POP3 does (I don't know if that would even be possible) but then the question remains, how do I do so? What settings do I have to touch to disable all the IMAP "improvements". There is a reason why (almost) all mail providers still support POP3 because it still has some value.
